autoload/lsp.vim

```diff
@@ -654,16 +654,17 @@ function! lsp#get_whitelisted_servers(...) abort
     return l:active_servers
 endfunction
 
-function! s:get_text_document_text(buf) abort
-    return join(getbufline(a:buf, 1, '$'), "\n")
+function! s:get_text_document_text(server_name, buf) abort
+    return join(s:get_last_file_content(a:server_name, a:buf), "\n")
```

Comment:
LSP protocol itself can handle `\n` or `\r\n` but do we want to explicitly send the correct line endings? Might be have a func in util that has `lsp#utils#line_ending(bufnr)`?

Also seems like we have removed. `s:requires_eol_at_eof`. More discussion about it can be found at https://github.com/prabirshrestha/vim-lsp/pull/158. It actually breaks pyls.

Reply:
If my understanding is correct, it's not required that the file content is same as server's one.